plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFactory.java

```diff
@@ -107,6 +107,10 @@ static Library create(IConfigurationElement configurationElement) throws Library
         MavenCoordinates mavenCoordinates = getMavenCoordinates(
             libraryFileElement.getChildren(ELEMENT_NAME_MAVEN_COORDINATES));
         LibraryFile libraryFile = loadSingleFile(libraryFileElement, mavenCoordinates);
+        // Pin servlet versions; update everything else
+        if (!libraryFile.getMavenCoordinates().getGroupId().startsWith("javax.servlet")) {
+          libraryFile.updateVersion();
+        }
```

Comment:
The library stuff always confuses me. I think this is not really effective, because when we call `Library.getAllDependencies()` on it, it will call `resolveDependencies()` which in turn calls `LibraryFile.updateVersion()`. Do you remember the recent test failure due to the Endpoints version update issue? Whichever version we specify in `plugin.xml`, I think they can eventually be upgraded.